docs/refactoring.md

```diff
@@ -0,0 +1,247 @@
+# Summary
```

Comment:
Remove? Might be nice to attach it to the PR description for posterity.

Reply:
I thought it could stay in the docs folder. Or did I read one of your earlier comments on this wrong?

Reply:
I don't mind it staying around for a short while. I thought you yourself said you'd remove it.

Reply:
I did, but then changed my mind :D